app/auth/views/suppliers.py

```diff
@@ -0,0 +1,30 @@
+from flask import jsonify
+from flask_login import current_user, login_required
+from app.auth import auth
+from app.utils import get_json_from_request, json_has_required_keys
+from app.auth.suppliers import get_supplier, update_supplier_details, valid_supplier
+
+
+@auth.route('/supplier', methods=['GET'])
+@login_required
+def get_supplier_profile():
+    if not valid_supplier(current_user):
+        return jsonify(message="Only users with role supplier can access supplier details"), 401
+    else:
+        supplier = get_supplier(current_user.supplier_code)
+        return jsonify(user=supplier), 200
```

Comment:
didn't Marco suggest this be "flat" rather than user.supplier

Reply:
you would pick it from state like `user: state:user`. There is no nesting after `user`

Reply:
ok